logger.go

```diff
@@ -47,30 +47,34 @@ type Logger struct {
 	callerSkip int
 }
 
-// New constructs a new Logger from the provided Facility and options. Passing
-// a nil Facility results in a no-op Logger.
-func New(fac zapcore.Facility, options ...Option) *Logger {
-	if fac == nil {
-		fac = zapcore.NopFacility()
+// New constructs a new Logger from the provided zapcore.Core and Options. Passing
+// a nil zapcore.Core returns a no-op Logger.
```

Comment:
instead of saying it's a noop logger, should we say we fallback to a noop core? (the logger itself is still functional in terms of calling hooks, still panics, etc)